### PR TITLE
Univariate regression

### PR DESCRIPTION
**Pull request by acopar**
Create univariate regression widget.

### CODE REVIEW

File: Orange/widgets/regression/owunivariateregression.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        self.legend.addItem(self.plot_item, 6*"&nbsp;" + "Regression Line")
+        self.update_legend()
+
+
+if __name__ == "__main__":
+    import sys
+    from PyQt4.QtGui import QApplication
+
+    a = QApplication(sys.argv)
+    ow = OWLinearRegression()
```
Comment (kernc) [inline Orange/widgets/regression/owunivariateregression.py]: test this.

**Comment by lanzagar** [comment]
A test for the new PolynomialLearner is missing (hence the failed coverage report).

**Comment by lanzagar** [comment]
1. widget is not resizeable
2. target class should be class var by default
3. **args is not needed
4. input learner type - all regressors?
5. handleNewSignals has apply - not needed in other setters?
6. In apply(): data_table creation from original table instead of numpy matrices

File: Orange/widgets/regression/owunivariateregression.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
+from PyQt4.QtGui import QLayout
+
+from Orange.data import Table, Domain
+from Orange.regression.linear import (RidgeRegressionLearner, LinearModel,
+                                      LinearRegressionLearner, PolynomialLearner)
+from Orange.regression.base_regression import SklLearnerRegression
+from Orange.preprocess.preprocess import Preprocess
+from Orange.widgets import widget, settings, gui
+from Orange.widgets.utils import itemmodels
```
Comment (lanzagar) [inline Orange/widgets/regression/owunivariateregression.py]: Orange imports should be the last (after third party imports such as PyQt and numpy)
https://www.python.org/dev/peps/pep-0008/#imports

File: Orange/regression/base_regression.py [hunk 3]
```diff
@@ -18,6 +18,5 @@ class ModelRegression(Model):
 class SklLearnerRegression(SklLearner, LearnerRegression):
     pass
 
-
```
Comment (lanzagar) [inline Orange/regression/base_regression.py]: https://www.python.org/dev/peps/pep-0008/#blank-lines

File: Orange/widgets/regression/owunivariateregression.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                                      LinearRegressionLearner, PolynomialLearner)
+from Orange.regression import Learner
+from Orange.preprocess.preprocess import Preprocess
+from Orange.widgets import widget, settings, gui
+from Orange.widgets.utils import itemmodels
+
+
+class OWUnivariateRegression(widget.OWWidget):
+    name = "Univariate Regression"
+    description = "Univariate regression algorithm with optional Linear " \
```
Comment (lanzagar) [inline Orange/widgets/regression/owunivariateregression.py]: How about just:
"Univariate regression with polynomial expansion."

File: Orange/regression/linear.py [hunk 5]
```diff
@@ -82,13 +83,33 @@ def __init__(self, loss='squared_loss', alpha=0.0001, epsilon=0.1,
...
+        sk = self.__wraps__(**self.params)
         clf = skl_pipeline.Pipeline(
             [('scaler', skl_preprocessing.StandardScaler()), ('sgd', sk)])
         clf.fit(X, Y.ravel())
         return LinearModel(clf)
 
 
+class PolynomialLearner(SklLearner):
+    __wraps__ = None
+    name = 'unireg'
```
Comment (lanzagar) [inline Orange/regression/linear.py]: Why is the name 'unireg'?
This is a general polynomial learner that can work with multiple variables.
Maybe 'poly learner' or even 'poly ' + learner.name?

File: Orange/tests/test_univariate_regression.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import unittest
+import numpy as np
+import Orange
+
+class UnivariateRegressionTest(unittest.TestCase):
```
Comment (lanzagar) [inline Orange/tests/test_univariate_regression.py]: This tests PolynomialLearner not univariate regression... (change name)

**Comment by lanzagar** [comment]
Replaced with #594

**Pull request closed without merging**